code/go/0chain.net/blobbercore/handler/object_operation_handler.go

```diff
@@ -1035,14 +1037,27 @@ func getFileOperation(r *http.Request) string {
 	return mode
 }
 
-func getExistingFileRef(fsh *StorageHandler, ctx context.Context, r *http.Request, allocationObj *allocation.Allocation, fileOperation string) *reference.Ref {
+//func getExistingFileRef(fsh *StorageHandler, ctx context.Context, r *http.Request, allocationObj *allocation.Allocation, fileOperation string) *reference.Ref {
+//	if fileOperation == constants.FileOperationInsert || fileOperation == constants.FileOperationUpdate {
+//		var formData allocation.UpdateFileChanger
+//		uploadMetaString := r.FormValue(getFormFieldName(fileOperation))
+//		err := json.Unmarshal([]byte(uploadMetaString), &formData)
+//
+//		if err == nil {
+//			return fsh.checkIfFileAlreadyExists(ctx, allocationObj.ID, formData.Path)
+//		}
+//	}
+//	return nil
+//}
+
+func getExistingFileRefID(fsh *StorageHandler, ctx context.Context, r *http.Request, allocationObj *allocation.Allocation, fileOperation string) *reference.Ref {
 	if fileOperation == constants.FileOperationInsert || fileOperation == constants.FileOperationUpdate {
 		var formData allocation.UpdateFileChanger
 		uploadMetaString := r.FormValue(getFormFieldName(fileOperation))
 		err := json.Unmarshal([]byte(uploadMetaString), &formData)
```

Comment:
@abouroumine If err is not nil then if better to return error rather than fileRef being nil.
I think its better to add `error` in return and return this error to the client